includes/HookHandlers/RequestWiki.php

```diff
@@ -2,118 +2,606 @@
-				'type' => 'text',
-				'section' => 'editing',
 				'hide-if' => [ '!==', 'edit-nsfw', '1' ],
+				'cssclass' => 'createwiki-infuse',
+				'section' => 'editing/request-info',
+				'required' => true,
+				'type' => 'text',
 				'default' => $wikiRequestManager->getExtraFieldData( 'nsfwtext' ),
 			]
 		);
 
-		RequestWikiFormUtils::insertFieldAfter(
+		RequestWikiFormUtils::addFieldToEnd(
 			$formDescriptor,
-			afterKey: 'edit-purpose',
 			newKey: 'edit-source',
 			newField: [
 				'label-message' => 'requestwiki-label-source',
+				'section' => 'editing/request-info',
 				'type' => 'check',
-				'section' => 'editing',
+				'cssclass' => 'createwiki-infuse',
 				'default' => $wikiRequestManager->getExtraFieldData( 'source' ),
 			]
 		);
 
-		RequestWikiFormUtils::insertFieldAfter(
+		RequestWikiFormUtils::addFieldToEnd(
 			$formDescriptor,
-			afterKey: 'edit-source',
 			newKey: 'edit-sourceurl',
 			newField: [
 				'label-message' => 'requestwiki-label-sourceurl',
-				'type' => 'url',
-				'section' => 'editing',
 				'hide-if' => [ '!==', 'edit-source', '1' ],
+				'cssclass' => 'createwiki-infuse',
+				'section' => 'editing/request-info',
+				'required' => true,
+				'type' => 'url',
 				'default' => $wikiRequestManager->getExtraFieldData( 'sourceurl' ),
 			]
 		);
 
+		RequestWikiFormUtils::addFieldToEnd(
+			$formDescriptor,
+			newKey: 'edit-defaultskin',
+			newField: [
+				'type' => 'select',
+				'label-message' => 'requestwiki-label-defaultskin',
+				'options' => array_combine(
+					$this->options->get( 'MirahezeMagicRequestWikiSkins' ),
+					$this->options->get( 'MirahezeMagicRequestWikiSkins' )
+				),
+				'section' => 'editing/configure',
+				'cssclass' => 'createwiki-infuse',
+				'default' => $wikiRequestManager->getExtraFieldData( 'defaultskin' ) ?? 'vector-2022',
+			]
+		);
+
+		RequestWikiFormUtils::addFieldToEnd(
+			$formDescriptor,
+			newKey: 'edit-wddescription',
+			newField: [
+				'type' => 'text',
+				'maxlength' => 512,
+				'label-message' => 'requestwiki-label-wddescription',
+				'section' => 'editing/configure',
+				'cssclass' => 'createwiki-infuse',
+				'default' => $wikiRequestManager->getExtraFieldData( 'wddescription' ),
+			]
+		);
+
+		$isAdvancedModified = $wikiRequestManager->getExtraFieldData( 'mainpageroot' ) !== false ||
+			$wikiRequestManager->getExtraFieldData( 'articlepath' ) !== '/wiki/$1' ||
+			$wikiRequestManager->getExtraFieldData( 'defaultextensions' ) !== [];
+
+		RequestWikiFormUtils::addFieldToEnd(
+			$formDescriptor,
+			newKey: 'edit-showadvanced',
+			newField: [
+				// We handle this manually, we don't want to save this field
+				'save' => false,
+				'class' => HTMLToggleSwitchField::class,
+				'cssclass' => 'createwiki-infuse',
+				'label-message' => 'requestwiki-label-showadvanced',
+				'section' => 'editing/advanced',
+				'default' => $isAdvancedModified,
+				'disabled' => $isAdvancedModified,
```

Comment:
Disabled?

Reply:
Yes so the button cant be toggled off if settings under it are on, this uses a dynamic toggle not a saved value, we don't want it turned off when saved if things under it are on or the settings also get lost. Its just a safeguard here.